example/copy/src/copy.cpp

```diff
@@ -0,0 +1,282 @@
+     * to memory on the desired device. Buffers can be
+     * allocated on the device or can be obtained from
+     * allready existing allocations e.g. std::array or
+     * std::vector.
+     **/
+    using Data = unsigned;
+    const Extents nElementsPerDim = 2;
+
+    const alpaka::Vec<DimMem, Size> extents(static_cast<Size>(nElementsPerDim),
+                                            static_cast<Size>(nElementsPerDim),
+                                            static_cast<Size>(nElementsPerDim));
+
+    std::array<Data, nElementsPerDim * nElementsPerDim * nElementsPerDim> plainBuffer;
+    alpaka::mem::buf::BufPlainPtrWrapper<DevHost, Data, DimMem, Size> hostBufferPlain(plainBuffer.data(), devHost, extents);
+    alpaka::mem::buf::Buf<DevHost, Data, DimMem, Size> hostBuffer (alpaka::mem::buf::alloc<Data, Size>(devHost, extents));
+    alpaka::mem::buf::Buf<DevAcc, Data, DimMem, Size> deviceBuffer(alpaka::mem::buf::alloc<Data, Size>(devAcc,  extents));
+
+
+    /**
+     * Init plain host buffer
+     *
+     * The buffer obtained from a plain pointer
+     * of the host is initialized here with the
+     * value zero. It is recommended to use a
+     * kernel for such a task, since a kernel
+     * can speed up the initialization process.
+     * The buffer can be provided to a kernel
+     * by passing the native pointer of the memory
+     * within the buffer (getPtrNative).
+     */
+    InitBufferKernel initBufferKernel;
+    Data initValue = 0;
+
+    auto const init (alpaka::exec::create<Host> (workdiv,
+                                                 initBufferKernel,
+                                                 alpaka::mem::view::getPtrNative(hostBufferPlain), // 1st kernel argument
+                                                 extents,                                          // 2nd kernel argument
+                                                 initValue));                                      // 3rd kernel argument
+    alpaka::stream::enqueue(stream, init);
+
+
+    /**
+     * Write some data to plain host buffer
+     *
+     * The buffer can not access the inner
+     * elements by some access operator
+     * directly, but it can return the
+     * pointer to the memory within the
+     * buffer (getPtrNative). This pointer
+     * can be used to write some values into
+     * the buffer memory. Mind, that only a host
+     * can write on host memory. The same holds
+     * for device memory.
+     */
+    for(size_t i = 0; i < extents.prod(); ++i){
+        alpaka::mem::view::getPtrNative(hostBufferPlain)[i] = static_cast<Data>(i);
+    }
+
+
+    /**
+     * Copy host to device Buffer
+     *
+     * A copy operation of one buffer into
+     * another buffer is enqueued into a stream
+     * like it is done for kernel execution, but
+     * more automatically. Copy is only possible
+     * from host to host, host to device and
+     * device to host. Device to device copies
```

Comment:
I think that this is not true.
I have never tested it, but you can copy from GPU to GPU. The GPU buffer holds the device handle to decide if it is a peer copy or a copy within a device.
Because there is only ever one CPU device shared by all CPU accelerators (at the moment), all CPU copie versions are the same.